drivers/sensor/bq34110/bq34110.c

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright (c) 2021 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_bq34110
+
+#include <init.h>
+#include <math.h>
+#include <string.h>
+#include <sys/util.h>
+#include <drivers/i2c.h>
+#include <sys/__assert.h>
+#include <sys/byteorder.h>
+#include <drivers/sensor.h>
+
+#include "bq34110.h"
+
+/* 5ms delay between each data flash write */
+#define BQ34110_DELAY	5
+/*
+ * @brief Read a register value
+ *
+ * Registers have an address and 16-bit value
+ *
+ * @param bq34110 Private data for the driver
+ * @param reg_addr Register address to read
+ * @param val Place to put the value on success
+ * @return 0 if success, or negative error code from I2C_API
+ */
+static int bq34110_cmd_reg_read(struct bq34110_data *bq34110,
+				uint8_t reg_addr, int16_t *val)
+{
+	int ret;
+	uint8_t i2c_data[2];
+
+	ret = i2c_write_read(bq34110->i2c, DT_INST_REG_ADDR(0), &reg_addr,
```

Comment:
I see register address 0 hardcoded here and in at least one other place. That doesn't look right; the address should be coming from the per-device information.